homeassistant/components/todoist/calendar.py

```diff
@@ -594,13 +594,14 @@ async def async_get_events(self, hass, start_date, end_date):
             )
             if not due_date:
                 continue
-            midnight = dt.as_utc(
-                dt.parse_datetime(
-                    due_date.strftime("%Y-%m-%d")
-                    + "T00:00:00"
-                    + self._api.state["user"]["tz_info"]["gmt_string"]
-                )
+            gmt_string = self._api.state["user"]["tz_info"]["gmt_string"]
+            local_midnight = dt.parse_datetime(
+                f'{due_date.strftime("%Y-%m-%d")}T00:00:00{gmt_string}'
             )
+            if local_midnight is not None:
+                midnight = dt.as_utc(local_midnight)
+            else:
+                midnight = due_date.replace(hour=0, minute=0, second=0, microsecond=0)
```

Comment:
Small change here since mypy correctly pointed out `dt.parse_datetime` could return None and `dt.as_utc` only allows a datetime instance.